### ISSUE TITLE
Fix js-bson v0.2.x for iojs

### ISSUE THREAD

**Issue by bitinn** [mongodb/js-bson#136]
- Too many mongodb modules still depends on mongodb driver 1.4.x
- And mongodb driver 1.4.x depends on bson 0.2.x
- And bson 0.2.x has set dependency on nan 1.7.0
- And following [your guide to debug c++ driver compile issue](http://christiankvalheim.com/post/diagnose_installation_problems/)

```
git clone https://github.com/mongodb/js-bson.git --branch V0.2.21
cd js-bson
npm install
make test
```

yield error

```
$ make test
node-gyp clean
gyp info it worked if it ends with ok
gyp info using node-gyp@2.0.2
gyp info using node@2.3.0 | darwin | x64
gyp info ok 
node-gyp configure build
gyp info it worked if it ends with ok
gyp info using node-gyp@2.0.2
gyp info using node@2.3.0 | darwin | x64
gyp info spawn python2
gyp info spawn args [ '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/teambegin/git/test/js-bson/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/teambegin/.node-gyp/2.3.0/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/teambegin/.node-gyp/2.3.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/node-gyp',
gyp info spawn args   '-Dmodule_root_dir=/Users/teambegin/git/test/js-bson',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/bson/ext/bson.o
In file included from ../ext/bson.cc:36:
In file included from ../ext/bson.h:17:
In file included from ../node_modules/nan/nan.h:80:
In file included from ../node_modules/nan/nan_new.h:190:
../node_modules/nan/nan_implementation_12_inl.h:181:66: error: too many
      arguments to function call, expected at most 2, have 4
  return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
         ~~~~~~~~~~~~~~~~~~                                      ^~~~~~~~~~
/Users/teambegin/.node-gyp/2.3.0/deps/v8/include/v8.h:4188:3: note: 'New'
      declared here
  static Local<Signature> New(
  ^
../ext/bson.cc:825:17: warning: 'DecodeBytes' is deprecated: Use
      DecodeBytes(isolate, ...) [-Wdeprecated-declarations]
                ssize_t len = DecodeBytes(args[0], BINARY);
                              ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:318:32: note: 'DecodeBytes' has been
      explicitly marked deprecated here
                inline ssize_t DecodeBytes(
                               ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
../ext/bson.cc:833:3: warning: 'DecodeWrite' is deprecated: Use
      DecodeWrite(isolate, ...) [-Wdeprecated-declarations]
                DecodeWrite(data, len, args[0], BINARY);
                ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:331:32: note: 'DecodeWrite' has been
      explicitly marked deprecated here
                inline ssize_t DecodeWrite(char* buf,
                               ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
../ext/bson.cc:928:28: warning: 'Encode' is deprecated: Use Encode(isolate, ...)
      [-Wdeprecated-declarations]
                Local<Value> bin_value = Encode(serialized_object, objec...
                                         ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:299:45: note: 'Encode' has been
      explicitly marked deprecated here
                inline v8::Local<v8::Value> Encode(
                                            ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
3 warnings and 1 error generated.
make[1]: *** [Release/obj.target/bson/ext/bson.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.4.0
gyp ERR! command "/usr/local/bin/iojs" "/usr/local/bin/node-gyp" "configure" "build"
gyp ERR! cwd /Users/teambegin/git/test/js-bson
gyp ERR! node -v v2.3.0
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok 
make: *** [node_gyp] Error 1
```

shrinkwrap build tree

```
...
        "mongodb": {
          "version": "1.4.38",
          "from": "mongodb@1.4.38",
          "resolved": "https://registry.npmjs.org/mongodb/-/mongodb-1.4.38.tgz",
          "dependencies": {
            "bson": {
              "version": "0.2.21",
              "from": "bson@>=0.2.0 <0.3.0",
              "resolved": "https://registry.npmjs.org/bson/-/bson-0.2.21.tgz",
              "dependencies": {
                "nan": {
                  "version": "1.7.0",
                  "from": "nan@1.7.0",
                  "resolved": "https://registry.npmjs.org/nan/-/nan-1.7.0.tgz"
                }
              }
            },
...
```

I would like to think this is an error that `nan` has fixed in later version, as mentioned here https://github.com/nodejs/nan/issues/343

Would much appreciate if you can help fix this issue @christkv !

**Comment by bitinn** [comment]
PS: this is also the problem I was having when I opened #130

**Comment by bitinn** [comment]
Just tested, replacing nan@1.7.0 with nan@1.8.4, and `make test` work just fine.

```
$ make test
node-gyp clean
gyp info it worked if it ends with ok
gyp info using node-gyp@2.0.2
gyp info using node@2.3.0 | darwin | x64
gyp info ok 
node-gyp configure build
gyp info it worked if it ends with ok
gyp info using node-gyp@2.0.2
gyp info using node@2.3.0 | darwin | x64
gyp info spawn python2
gyp info spawn args [ '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/teambegin/git/test/js-bson/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/teambegin/.node-gyp/2.3.0/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/teambegin/.node-gyp/2.3.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/node-gyp',
gyp info spawn args   '-Dmodule_root_dir=/Users/teambegin/git/test/js-bson',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/bson/ext/bson.o
../ext/bson.cc:825:17: warning: 'DecodeBytes' is deprecated: Use
      DecodeBytes(isolate, ...) [-Wdeprecated-declarations]
                ssize_t len = DecodeBytes(args[0], BINARY);
                              ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:318:32: note: 'DecodeBytes' has been
      explicitly marked deprecated here
                inline ssize_t DecodeBytes(
                               ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
../ext/bson.cc:833:3: warning: 'DecodeWrite' is deprecated: Use
      DecodeWrite(isolate, ...) [-Wdeprecated-declarations]
                DecodeWrite(data, len, args[0], BINARY);
                ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:331:32: note: 'DecodeWrite' has been
      explicitly marked deprecated here
                inline ssize_t DecodeWrite(char* buf,
                               ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
../ext/bson.cc:928:28: warning: 'Encode' is deprecated: Use Encode(isolate, ...)
      [-Wdeprecated-declarations]
                Local<Value> bin_value = Encode(serialized_object, objec...
                                         ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:299:45: note: 'Encode' has been
      explicitly marked deprecated here
                inline v8::Local<v8::Value> Encode(
                                            ^
/Users/teambegin/.node-gyp/2.3.0/src/node.h:66:42: note: expanded from macro
      'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                                         ^
3 warnings generated.
  SOLINK_MODULE(target) Release/bson.node
gyp info ok 
npm test

> bson@0.2.21 test /Users/teambegin/git/test/js-bson
> nodeunit ./test/node && TEST_NATIVE=TRUE nodeunit ./test/node


bson_array_test
✔ shouldCorrectlyDeserializeArray
✔ shouldCorrectlySerializeUsingTypedArray
✔ noGlobalsLeaked

bson_parser_comparision_test
✔ Should Correctly Serialize and Deserialize simple edge value
✔ Should Correctly execute toJSON
✔ Should Serialize and Deserialize nested document
✔ Simple integer serialization/deserialization test, including testing boundary conditions
✔ Simple serialization and deserialization test for a Long value
✔ Simple serialization and deserialization for a Float value
✔ Simple serialization and deserialization for a null value
✔ Simple serialization and deserialization for a boolean value
✔ Simple serialization and deserialization for a date value
✔ Simple serialization and deserialization for a objectId value
✔ Simple serialization and deserialization for a Binary value
✔ Simple serialization and deserialization for a Binary value of type 2
✔ Complex nested serialization and deserialization for a Code value
✔ Simple serialization and deserialization for a Code value
✔ Simple serialization and deserialization for an Object
✔ Simple serialization and deserialization for a DBRef
✔ Serialize utf8
✔ Serialize object with array
✔ Test equality of an object ID
✔ Test same serialization for Object ID
✔ Complex object serialization
✔ Serialize function
✔ Serialize document with special operators
✔ Create ObjectID from hex string
✔ Serialize big complex document
✔ Should Correctly Serialize undefined as null and Deserialize it as null value
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly Serialize and Deserialize Number 2
✔ Should Correctly Correctly handle Long promotion
✔ Should error out due to 24 characters but not valid hexstring for ObjectID
✔ Should correctly serialize prototype fields
✔ Should correctly serialize regexp
✔ noGlobalsLeaked

bson_test
✔ Should Correctly create ObjectID and do deep equals
✔ Should Correctly convert ObjectID to itself
✔ Should Correctly get BSON types from require
✔ Should Correctly Deserialize object
✔ Should Correctly Deserialize object with all types
✔ Should Serialize and Deserialize String
✔ Should Serialize and Deserialize Empty String
✔ Should Correctly Serialize and Deserialize Integer
✔ Should Correctly Serialize and Deserialize null value
✔ Should Correctly Serialize and Deserialize Number 1
✔ Should Correctly Serialize and Deserialize Object
✔ Should Correctly Serialize undefined as null and Deserialize it as null value
✔ Should Correctly Serialize and Deserialize Array
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly Serialize and Deserialize Number 4
✔ Should Correctly Serialize and Deserialize Array with added on functions
✔ Should correctly deserialize a nested object
✔ Should Correctly Serialize and Deserialize A Boolean
✔ Should Correctly Serialize and Deserialize a Date
✔ Should Correctly Serialize and Deserialize a Date from another VM
✔ Should Correctly Serialize nested doc
✔ Should Correctly Serialize and Deserialize Oid
✔ Should Correctly encode Empty Hash
✔ Should Correctly Serialize and Deserialize Ordered Hash
✔ Should Correctly Serialize and Deserialize Regular Expression
✔ Should Correctly Serialize and Deserialize a Binary object
✔ Should Correctly Serialize and Deserialize a Type 2 Binary object
.write(string, encoding, offset, length) is deprecated. Use write(string[, offset[, length]][, encoding]) instead.
✔ Should Correctly Serialize and Deserialize a big Binary object
✔ Should Correctly Serialize and Deserialize DBRef
✔ Should Correctly Serialize and Deserialize partial DBRef
✔ Should Correctly Serialize and Deserialize simple Int
✔ Should Correctly Serialize and Deserialize Long Integer
✔ Should Deserialize Large Integers as Number not Long
✔ Should Correctly Serialize and Deserialize Long Integer and Timestamp as different types
✔ Should Always put the id as the first item in a hash
✔ Should Correctly Serialize and Deserialize a User defined Binary object
✔ Should Correclty Serialize and Deserialize a Code object
✔ Should Correctly serialize and deserialize and embedded array
✔ Should Correctly Serialize and Deserialize UTF8
✔ Should Correctly Serialize and Deserialize query object
✔ Should Correctly Serialize and Deserialize empty query object
✔ Should Correctly Serialize and Deserialize array based doc
✔ Should Correctly Serialize and Deserialize Symbol
✔ Should handle Deeply nested document
✔ Should handle complicated all typed object
✔ Should Correctly Serialize Complex Nested Object
✔ Should correctly massive doc
✔ Should Correctly Serialize/Deserialize regexp object
✔ Should Correctly Serialize/Deserialize complicated object
✔ Should Correctly Serialize/Deserialize nested object
✔ Should Correctly Serialize/Deserialize nested object with even more nesting
✔ Should Correctly Serialize empty name object
✔ Should Correctly handle Forced Doubles to ensure we allocate enough space for cap collections
✔ Should deserialize correctly
✔ Should correctly serialize and deserialize MinKey and MaxKey values
✔ Should correctly serialize Double value
✔ ObjectID should correctly create objects
✔ ObjectID should correctly retrieve timestamp
✔ Should Correctly throw error on bsonparser errors
✔ Should correctly calculate the size of a given javascript object
✔ Should correctly calculate the size of a given javascript object using instance method
✔ Should correctly serializeWithBufferAndIndex a given javascript object
✔ Should correctly serializeWithBufferAndIndex a given javascript object using a BSON instance
✔ Should correctly serialize a given javascript object
✔ Should correctly serialize a given javascript object using a bson instance
✔ Should correctly deserialize a buffer using the BSON class level parser
✔ Should correctly deserialize a buffer using the BSON instance parser
✔ Should correctly deserializeStream a buffer object
✔ ObjectID should have a correct cached representation of the hexString
✔ Should fail to create ObjectID due to illegal hex code
✔ noGlobalsLeaked

bson_typed_array_test
✔ shouldCorrectlyDeserializeUsingTypedArray
✔ shouldCorrectlySerializeUsingTypedArray
✔ exercise all the binary object constructor methods
✔ exercise the put binary object method for an instance when using Uint8Array
✔ exercise the write binary object method for an instance when using Uint8Array
✔ exercise the read binary object method for an instance when using Uint8Array
✔ noGlobalsLeaked

test_full_bson
✔ Should Correctly Deserialize object with all types
✔ Should Serialize and Deserialize String
✔ Should Correctly Serialize and Deserialize Integer
✔ Should Correctly Serialize and Deserialize null value
✔ Should Correctly Serialize and Deserialize undefined value
✔ Should Correctly Serialize and Deserialize Number 3
✔ Should Correctly Serialize and Deserialize Object
✔ Should Correctly Serialize and Deserialize Array
✔ Should Correctly Serialize and Deserialize Array with added on functions
✔ Should Correctly Serialize and Deserialize A Boolean
✔ Should Correctly Serialize and Deserialize a Date
✔ Should Correctly Serialize and Deserialize Oid
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly encode Empty Hash
✔ Should Correctly Serialize and Deserialize Ordered Hash
✔ Should Correctly Serialize and Deserialize Regular Expression
✔ Should Correctly Serialize and Deserialize a Binary object
✔ Should Correctly Serialize and Deserialize a big Binary object
✔ Should Correctly fail due to attempting serialization of illegal key values
✔ noGlobalsLeaked

to_bson_test
✔ Should correctly handle toBson function for an object
✔ Should correctly handle embedded toBson function for an object
✔ Should correctly serialize when embedded non object returned by toBSON
✔ Should fail when top level object returns a non object type
✔ noGlobalsLeaked

OK: 55958 assertions (281ms)
TAP version 13
1..0
# time=339.734ms

bson_array_test
✔ shouldCorrectlyDeserializeArray
✔ shouldCorrectlySerializeUsingTypedArray
✔ noGlobalsLeaked

bson_parser_comparision_test
✔ Should Correctly Serialize and Deserialize simple edge value
✔ Should Correctly execute toJSON
✔ Should Serialize and Deserialize nested document
✔ Simple integer serialization/deserialization test, including testing boundary conditions
✔ Simple serialization and deserialization test for a Long value
✔ Simple serialization and deserialization for a Float value
✔ Simple serialization and deserialization for a null value
✔ Simple serialization and deserialization for a boolean value
✔ Simple serialization and deserialization for a date value
✔ Simple serialization and deserialization for a objectId value
✔ Simple serialization and deserialization for a Binary value
✔ Simple serialization and deserialization for a Binary value of type 2
✔ Complex nested serialization and deserialization for a Code value
✔ Simple serialization and deserialization for a Code value
✔ Simple serialization and deserialization for an Object
✔ Simple serialization and deserialization for a DBRef
✔ Serialize utf8
✔ Serialize object with array
✔ Test equality of an object ID
✔ Test same serialization for Object ID
✔ Complex object serialization
✔ Serialize function
✔ Serialize document with special operators
✔ Create ObjectID from hex string
✔ Serialize big complex document
✔ Should Correctly Serialize undefined as null and Deserialize it as null value
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly Serialize and Deserialize Number 2
✔ Should Correctly Correctly handle Long promotion
✔ Should error out due to 24 characters but not valid hexstring for ObjectID
✔ Should correctly serialize prototype fields
✔ Should correctly serialize regexp
✔ noGlobalsLeaked

bson_test
✔ Should Correctly create ObjectID and do deep equals
✔ Should Correctly convert ObjectID to itself
✔ Should Correctly get BSON types from require
✔ Should Correctly Deserialize object
✔ Should Correctly Deserialize object with all types
✔ Should Serialize and Deserialize String
✔ Should Serialize and Deserialize Empty String
✔ Should Correctly Serialize and Deserialize Integer
✔ Should Correctly Serialize and Deserialize null value
✔ Should Correctly Serialize and Deserialize Number 1
✔ Should Correctly Serialize and Deserialize Object
✔ Should Correctly Serialize undefined as null and Deserialize it as null value
✔ Should Correctly Serialize and Deserialize Array
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly Serialize and Deserialize Number 4
✔ Should Correctly Serialize and Deserialize Array with added on functions
✔ Should correctly deserialize a nested object
✔ Should Correctly Serialize and Deserialize A Boolean
✔ Should Correctly Serialize and Deserialize a Date
✔ Should Correctly Serialize and Deserialize a Date from another VM
✔ Should Correctly Serialize nested doc
✔ Should Correctly Serialize and Deserialize Oid
✔ Should Correctly encode Empty Hash
✔ Should Correctly Serialize and Deserialize Ordered Hash
✔ Should Correctly Serialize and Deserialize Regular Expression
✔ Should Correctly Serialize and Deserialize a Binary object
✔ Should Correctly Serialize and Deserialize a Type 2 Binary object
.write(string, encoding, offset, length) is deprecated. Use write(string[, offset[, length]][, encoding]) instead.
✔ Should Correctly Serialize and Deserialize a big Binary object
✔ Should Correctly Serialize and Deserialize DBRef
✔ Should Correctly Serialize and Deserialize partial DBRef
✔ Should Correctly Serialize and Deserialize simple Int
✔ Should Correctly Serialize and Deserialize Long Integer
✔ Should Deserialize Large Integers as Number not Long
✔ Should Correctly Serialize and Deserialize Long Integer and Timestamp as different types
✔ Should Always put the id as the first item in a hash
✔ Should Correctly Serialize and Deserialize a User defined Binary object
✔ Should Correclty Serialize and Deserialize a Code object
✔ Should Correctly serialize and deserialize and embedded array
✔ Should Correctly Serialize and Deserialize UTF8
✔ Should Correctly Serialize and Deserialize query object
✔ Should Correctly Serialize and Deserialize empty query object
✔ Should Correctly Serialize and Deserialize array based doc
✔ Should Correctly Serialize and Deserialize Symbol
✔ Should handle Deeply nested document
✔ Should handle complicated all typed object
✔ Should Correctly Serialize Complex Nested Object
✔ Should correctly massive doc
✔ Should Correctly Serialize/Deserialize regexp object
✔ Should Correctly Serialize/Deserialize complicated object
✔ Should Correctly Serialize/Deserialize nested object
✔ Should Correctly Serialize/Deserialize nested object with even more nesting
✔ Should Correctly Serialize empty name object
✔ Should Correctly handle Forced Doubles to ensure we allocate enough space for cap collections
✔ Should deserialize correctly
✔ Should correctly serialize and deserialize MinKey and MaxKey values
✔ Should correctly serialize Double value
✔ ObjectID should correctly create objects
✔ ObjectID should correctly retrieve timestamp
✔ Should Correctly throw error on bsonparser errors
✔ Should correctly calculate the size of a given javascript object
✔ Should correctly calculate the size of a given javascript object using instance method
✔ Should correctly serializeWithBufferAndIndex a given javascript object
✔ Should correctly serializeWithBufferAndIndex a given javascript object using a BSON instance
✔ Should correctly serialize a given javascript object
✔ Should correctly serialize a given javascript object using a bson instance
✔ Should correctly deserialize a buffer using the BSON class level parser
✔ Should correctly deserialize a buffer using the BSON instance parser
✔ Should correctly deserializeStream a buffer object
✔ ObjectID should have a correct cached representation of the hexString
✔ Should fail to create ObjectID due to illegal hex code
✔ noGlobalsLeaked

bson_typed_array_test
✔ shouldCorrectlyDeserializeUsingTypedArray
✔ shouldCorrectlySerializeUsingTypedArray
✔ exercise all the binary object constructor methods
✔ exercise the put binary object method for an instance when using Uint8Array
✔ exercise the write binary object method for an instance when using Uint8Array
✔ exercise the read binary object method for an instance when using Uint8Array
✔ noGlobalsLeaked

test_full_bson
✔ Should Correctly Deserialize object with all types
✔ Should Serialize and Deserialize String
✔ Should Correctly Serialize and Deserialize Integer
✔ Should Correctly Serialize and Deserialize null value
✔ Should Correctly Serialize and Deserialize undefined value
✔ Should Correctly Serialize and Deserialize Number 3
✔ Should Correctly Serialize and Deserialize Object
✔ Should Correctly Serialize and Deserialize Array
✔ Should Correctly Serialize and Deserialize Array with added on functions
✔ Should Correctly Serialize and Deserialize A Boolean
✔ Should Correctly Serialize and Deserialize a Date
✔ Should Correctly Serialize and Deserialize Oid
✔ Should Correctly Serialize and Deserialize Buffer
✔ Should Correctly encode Empty Hash
✔ Should Correctly Serialize and Deserialize Ordered Hash
✔ Should Correctly Serialize and Deserialize Regular Expression
✔ Should Correctly Serialize and Deserialize a Binary object
✔ Should Correctly Serialize and Deserialize a big Binary object
✔ Should Correctly fail due to attempting serialization of illegal key values
✔ noGlobalsLeaked

to_bson_test
✔ Should correctly handle toBson function for an object
✔ Should correctly handle embedded toBson function for an object
✔ Should correctly serialize when embedded non object returned by toBSON
✔ Should fail when top level object returns a non object type
✔ noGlobalsLeaked

OK: 55963 assertions (280ms)
TAP version 13
1..0
# time=337.297ms
```

**Comment by christkv** [comment]
I published a 0.2.22 so if you reinstall the 1.4.x branch it should pull the latest bson module in the 0.2.x series and build correctly

**Comment by bitinn** [comment]
@christkv thx!
